x-pack/plugins/lens/public/xy_visualization/axis_settings_popover.tsx

```diff
@@ -0,0 +1,244 @@
+}
+const popoverConfig = (
+  axis: AxesSettingsConfigKeys
+): { icon: IconType; groupPosition: ToolbarButtonProps['groupPosition'] } => {
+  switch (axis) {
+    case 'yLeft':
+      return {
+        icon: EuiIconAxisLeft,
+        groupPosition: 'left',
+      };
+    case 'yRight':
+      return {
+        icon: EuiIconAxisRight,
+        groupPosition: 'right',
+      };
+    case 'x':
+    default:
+      return {
+        icon: EuiIconAxisBottom,
+        groupPosition: 'center',
+      };
+  }
+};
+export const AxisSettingsPopover: React.FunctionComponent<AxisSettingsPopoverProps> = ({
+  layers,
+  popoverTitle,
+  axis,
+  axisTitle,
+  frame,
+  updateTitleState,
+  toggleTickLabelsVisibility,
+  toggleGridlinesVisibility,
+  isDisabled,
+  areTickLabelsVisible,
+  areGridlinesVisible,
+  isAxisTitleVisible,
+  toggleAxisTitleVisibility,
+}) => {
+  const [popoversOpenState, setPopoversOpenState] = useState(false);
+  const [title, setTitle] = useState<string | undefined>(axisTitle);
+
+  const getAxisTitle = useCallback(() => {
+    const defaultTitle = axisTitle;
+
+    const initialLayer = layers?.[0];
+    if (!initialLayer || !initialLayer.accessors.length) {
+      return defaultTitle;
+    }
+    const datasource = frame.datasourceLayers[initialLayer.layerId];
+    if (!datasource) {
+      return defaultTitle;
+    }
+    switch (axis) {
+      case 'yLeft':
+        const yLeft = initialLayer.accessors[0]
+          ? datasource.getOperationForColumnId(initialLayer.accessors[0])
+          : null;
+        return defaultTitle || yLeft?.label;
+      case 'x':
+      default:
+        const x = initialLayer.xAccessor
+          ? datasource.getOperationForColumnId(initialLayer.xAccessor)
+          : null;
+        return defaultTitle || x?.label;
+      case 'yRight':
+        let yRightTitle: string = '';
+        const rightAxisLayer = layers?.filter(
+          (layer) =>
+            layer.yConfig && layer.yConfig.length > 0 && layer.yConfig[0].axisMode === 'right'
+        );
+        if (rightAxisLayer && rightAxisLayer.length > 0) {
+          const yConfig = rightAxisLayer[0].yConfig;
+          const dataSourceNewLayer = frame.datasourceLayers[rightAxisLayer[0].layerId];
+          const y = yConfig
+            ? dataSourceNewLayer.getOperationForColumnId(yConfig[0].forAccessor)
+            : null;
+          yRightTitle = y?.label || '';
+        }
+        return defaultTitle || yRightTitle;
+    }
+    /* We want this callback to run only if open changes its state. What we want to accomplish here is to give the user a better UX.
+       By default these input fields have the axis legends. If the user changes the input text, the axis legends should also change.
+       BUT if the user cleans up the input text, it should remain empty until the user closes and reopens the panel.
+       In that case, the default axis legend should appear. */
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Your reason for disabling this linting rule doesn't match the purpose of the linting rule: the main reason it's here is to prevent weird scoping issues due to the way react evaluates hooks. It looks like this function would run quickly in all cases, does it even need a `useCallback`?

Reply:
We want this function to run only when popover changes its open state. This is the reason this rule is commented out. It is due to the UX we want to give to the user. It should not be updated if anything else changes, only when the popover open state changes.